### PR TITLE
fix date param type map

### PR DESCRIPTION
**Pull request by huayang-codaio**
https://staging.coda.io/d/Packs-IA-go-packs_d36WK4zgrYx/Bugs-Feature-requests_suyN0#Open-Bugs_tu6FU/r115

param type of date should be mapped to string when it's passed to execute method. 

`TypeMap` was used by both params and results. but it seems like at the moment we are using `ValueType` for result type. `TypeMap` is only used to type `PackFormulaResult` which isn't precise anyway (it doesn't respect `formula.resultType` right now). So it looks like fixing param type of date doesn't have any further side effects. 

I have no idea of the generated doc files. but they are generated in main.

### CODE REVIEW

File: api_types.ts [hunk 1]
```diff
@@ -62,13 +62,13 @@ interface TypeMap {
...
   [Type.boolean]: boolean;
-  [Type.date]: Date;
+  [Type.date]: string;
   [Type.html]: string;
   [Type.image]: string;
 }
 
 export type PackFormulaValue = $Values<Omit<TypeMap, Type.object>> | PackFormulaValue[];
-export type PackFormulaResult = $Values<TypeMap> | PackFormulaResult[];
+export type PackFormulaResult = $Values<Omit<TypeMap, Type.date>> | PackFormulaResult[];
```
Comment (jonathan-codaio) [inline api_types.ts:71]: Hmm, is this saying that if you try to use `makeFormula` to define a formula that returns `Type.date` it won't compile? Seems like up until now we technically supported that case, right?

If so, was hoping we wouldn't be disallowing that, and could somehow reconcile dates as parameters being strings while dates as return values still being Dates.

File: api_types.ts [hunk 2]
```diff
@@ -62,13 +62,13 @@ interface TypeMap {
   [Type.string]: string;
   [Type.object]: object;
   [Type.boolean]: boolean;
-  [Type.date]: Date;
+  [Type.date]: string;
```
Comment (jonathan-codaio) [inline api_types.ts:65]: Do you know if this affects when a date is used in an object schema property? I think that's unrelated but I forget sometimes.

**Review by jonathan-codaio — COMMENTED** [review]
Sorry about the documentation diffs, not sure why my last PR removed them, but it looks like they come back easily enough here: https://github.com/coda/packs-sdk/pull/1333

File: api_types.ts (hunk 1 above)
Comment (jonathan-codaio) [reply api_types.ts:71]: Oh you're right, we changed `makeFormula()` to require declaring a primitive type as the result type (ValueType) and if you want to get a rich value like a date you have to use codaType (ValueHintType). So this is probably fine.

Can you just patch this in to `coda` and `packs` and make sure there aren't any TS errors with this change?

File: api_types.ts (hunk 2 above)
Comment (huayang-codaio) [reply api_types.ts:65]: good point. let me try that.

**Comment by huayang-codaio** [comment]
Closing this one. Confirmed with Eric that no repro.

**Comment by jonathan-codaio** [comment]
Fascinating. Do we have any theories on where the serialized formula params get converted back into Dates?

**Comment by huayang-codaio** [comment]
I spent some time reading the code but it was not clear to me where the conversion happens.

**Comment by jonathan-codaio** [comment]
Oh here we go: https://github.com/coda/coda/blob/main/modules/protos/protos/helpers/calc_service.ts#L42

Looks like we marshal from the grpc request into JS objects based on the declared types. For v1 execution, those JS objects will be used as-is. For v2, your new marshaling will make sure a Date stays a Date.

**Comment by huayang-codaio** [comment]
ah, thanks for the quick find. I was searching for `Date.parse` lol 

but I suspect it's also a Date for v1 executions. otherwise formulas like google calendar events sync would not be able to execute, see below (`toISOString` isn't a function on strings).

```
    execute: async ([dateRange, calendar, search, showDeleted, showHiddenInvitations, excludeDeclined], context) => {
      const [minDate, maxDate] = dateRange;
      const {continuation} = context.sync;
      const calendarId = await helpers.getCalendarId(calendar, context);

      const params = objectUtils.removeEmpty({
        timeMin: minDate.toISOString(),
        timeMax: maxDate.toISOString(),
        singleEvents: true,
        orderBy: 'startTime',
        pageToken: continuation?.nextPageToken,
        q: search,
        showDeleted,
        showHiddenInvitations,
      });
```

**Comment by huayang-codaio** [comment]
oh, maybe your code is pointing v1 executions using Date. ignore more previous comment in that case.

**Comment by jonathan-codaio** [comment]
Yeah it's Date in both cases. Was just saying that after the proto value is marshaled to a Date, in the v1 case it stays a date because that post-marshaled value is used as-is when executed in the calc server. In the v2 case it stays a date because your new lambda marshaling code ensures that it does even while making the extra hops to lambda and IVM.

**Comment by huayang-codaio** [comment]
cool cool. thanks for explaining!

**Pull request closed without merging**